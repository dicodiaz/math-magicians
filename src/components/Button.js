import PropTypes from 'prop-types';
import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function
export default class Button extends React.Component {
  render() {
    const { classes, text } = this.props;
    classes.push('py-4', 'border', 'border-custom1', 'fs-5');
    const classesString = classes.join(' ');
    return (
      <button type="button" className={classesString}>
        {text}
      </button>
    );
  }
}

Button.propTypes = {
  classes: PropTypes.arrayOf(PropTypes.string).isRequired,
  text: PropTypes.string.isRequired,
};
