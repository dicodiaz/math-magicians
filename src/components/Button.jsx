import PropTypes from 'prop-types';

const Button = ({ text, type, onClick }) => {
  const classNameArray = ['py-4', 'border', 'border-custom1', 'fs-5'];
  switch (type) {
    case 'normal':
      classNameArray.push('bg-custom2');
      break;
    case 'operator':
      classNameArray.push('bg-custom3');
      break;
    case 'wide':
      classNameArray.push('bg-custom2', 'col-6');
      break;
    default:
  }
  const className = classNameArray.join(' ');

  return (
    <button type="button" className={className} onClick={onClick}>
      {text}
    </button>
  );
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
