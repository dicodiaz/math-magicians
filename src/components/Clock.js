import PropTypes from 'prop-types';
import React from 'react';

export default class Clock extends React.Component {
  constructor(props) {
    super(props);
    this.state = { date: new Date() };
  }

  componentDidMount() {
    const { updateEveryNSeconds } = this.props;
    this.timerID = setInterval(() => this.tick(), 1000 * updateEveryNSeconds);
  }

  componentWillUnmount() {
    clearInterval(this.timerID);
  }

  tick() {
    this.setState({
      date: new Date(),
    });
  }

  render() {
    const { date } = this.state;
    return (
      <div className="col-md-9">
        <div className="row text-end text-white">
          <p>{date.toLocaleTimeString()}</p>
        </div>
      </div>
    );
  }
}

Clock.propTypes = {
  updateEveryNSeconds: PropTypes.number.isRequired,
};
