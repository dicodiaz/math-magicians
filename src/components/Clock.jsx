import PropTypes from 'prop-types';
import { useEffect, useState } from 'react';

const Clock = ({ updateEveryNSeconds }) => {
  const [date, setDate] = useState(new Date());

  useEffect(() => {
    const timerID = setInterval(() => setDate(new Date()), 1000 * updateEveryNSeconds);
    return () => clearInterval(timerID);
  }, [updateEveryNSeconds]);

  return (
    <div className="col-md-9">
      <div className="row text-end text-white">
        <p data-testid="time">{date.toLocaleTimeString()}</p>
      </div>
    </div>
  );
};

Clock.propTypes = {
  updateEveryNSeconds: PropTypes.number.isRequired,
};

export default Clock;
