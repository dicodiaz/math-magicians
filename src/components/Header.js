import { AiOutlineMenu } from 'react-icons/ai';
import { Link } from 'react-router-dom';

const Header = () => (
  <nav className="navbar navbar-expand-md navbar-light bg-secondary fixed-top">
    <div className="container-fluid">
      <span className="navbar-brand text-white">Math Magicians</span>
      <button
        className="navbar-toggler"
        type="button"
        data-bs-toggle="collapse"
        data-bs-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <AiOutlineMenu style={{ color: 'white', fontSize: '25px' }} />
      </button>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav ms-auto">
          <li className="nav-item">
            <Link to="/">Home</Link>
          </li>
          <li className="v-line d-none d-md-block" />
          <li className="nav-item">
            <Link to="/calculator">Calculator</Link>
          </li>
          <li className="v-line d-none d-md-block" />
          <li className="nav-item">
            <Link to="/quote">Quote</Link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);

export default Header;
