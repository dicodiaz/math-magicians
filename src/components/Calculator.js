import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Button from './Button';

const Calculator = () => {
  const [obj, setObj] = useState({ total: '0', next: null, operation: null });
  const { total, next, operation } = obj;

  const handleClick = (text) => {
    setObj(calculate(obj, text));
  };

  return (
    <div className="col-md-9">
      <div className="row mx-0 row-cols-4 text-center">
        <div className="bg-custom1 text-white text-end col-12 py-4 fs-5">
          {total}
          {operation}
          {next}
        </div>
        <Button classes={['bg-custom2']} text="AC" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="+/-" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="%" handleClick={handleClick} />
        <Button classes={['bg-custom3']} text="&divide;" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="7" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="8" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="9" handleClick={handleClick} />
        <Button classes={['bg-custom3']} text="&times;" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="4" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="5" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="6" handleClick={handleClick} />
        <Button classes={['bg-custom3']} text="&minus;" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="1" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="2" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="3" handleClick={handleClick} />
        <Button classes={['bg-custom3']} text="+" handleClick={handleClick} />
        <Button classes={['bg-custom2', 'col-6']} text="0" handleClick={handleClick} />
        <Button classes={['bg-custom2']} text="." handleClick={handleClick} />
        <Button classes={['bg-custom3']} text="=" handleClick={handleClick} />
      </div>
    </div>
  );
};

export default Calculator;
