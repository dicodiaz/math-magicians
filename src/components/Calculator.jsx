import { useState } from 'react';
import buttons from '../data/buttons';
import calculate from '../logic/calculate';
import Button from './Button';

const Calculator = () => {
  const [obj, setObj] = useState({ total: '0', next: null, operation: null });
  const { total, next, operation } = obj;

  const handleClick = (e) => {
    setObj(calculate(obj, e.target.textContent));
  };

  return (
    <div className="col-md-9 pt-5">
      <div className="row mx-0 g-0 row-cols-4 text-center">
        <div data-testid="screen" className="bg-custom1 text-white text-end col-12 p-4 fs-5">
          {total}
          {operation}
          {next}
        </div>
        {buttons.map((button) => {
          const { id, text, type } = button;
          return <Button key={id} text={text} type={type} onClick={handleClick} />;
        })}
      </div>
    </div>
  );
};

export default Calculator;
