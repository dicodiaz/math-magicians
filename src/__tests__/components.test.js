import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Button from '../components/Button';
import Calculator from '../components/Calculator';
import Clock from '../components/Clock';
import Header from '../components/Header';

describe('testing components', () => {
  test('testing Button.js', () => {
    const tree = render(<Button classes={['bg-custom2']} text="AC" handleClick={(text) => text} />);
    expect(tree).toMatchSnapshot();
  });

  test('testing Calculator.js', () => {
    const tree = render(<Calculator />);
    expect(tree).toMatchSnapshot();
  });

  test('testing Clock.js', () => {
    const tree = render(<Clock updateEveryNSeconds={2} />);
    expect(tree).toMatchSnapshot();
  });

  test('testing Header.js', () => {
    const tree = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });
});
