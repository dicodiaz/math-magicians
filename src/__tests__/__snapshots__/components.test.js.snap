// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing components testing Button.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="bg-custom2 py-4 border border-custom1 fs-5"
        type="button"
      >
        AC
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="bg-custom2 py-4 border border-custom1 fs-5"
      type="button"
    >
      AC
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing components testing Calculator.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-md-9"
      >
        <div
          class="row mx-0 row-cols-4 text-center"
        >
          <div
            class="bg-custom1 text-white text-end col-12 py-4 fs-5"
            data-testid="screen"
          >
            0
          </div>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            AC
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            +/-
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            %
          </button>
          <button
            class="bg-custom3 py-4 border border-custom1 fs-5"
            type="button"
          >
            ÷
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            7
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            8
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            9
          </button>
          <button
            class="bg-custom3 py-4 border border-custom1 fs-5"
            type="button"
          >
            ×
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            4
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            5
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            6
          </button>
          <button
            class="bg-custom3 py-4 border border-custom1 fs-5"
            type="button"
          >
            −
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            1
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            2
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            3
          </button>
          <button
            class="bg-custom3 py-4 border border-custom1 fs-5"
            type="button"
          >
            +
          </button>
          <button
            class="bg-custom2 col-6 py-4 border border-custom1 fs-5"
            type="button"
          >
            0
          </button>
          <button
            class="bg-custom2 py-4 border border-custom1 fs-5"
            type="button"
          >
            .
          </button>
          <button
            class="bg-custom3 py-4 border border-custom1 fs-5"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-md-9"
    >
      <div
        class="row mx-0 row-cols-4 text-center"
      >
        <div
          class="bg-custom1 text-white text-end col-12 py-4 fs-5"
          data-testid="screen"
        >
          0
        </div>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          AC
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          +/-
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          %
        </button>
        <button
          class="bg-custom3 py-4 border border-custom1 fs-5"
          type="button"
        >
          ÷
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          7
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          8
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          9
        </button>
        <button
          class="bg-custom3 py-4 border border-custom1 fs-5"
          type="button"
        >
          ×
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          4
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          5
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          6
        </button>
        <button
          class="bg-custom3 py-4 border border-custom1 fs-5"
          type="button"
        >
          −
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          1
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          2
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          3
        </button>
        <button
          class="bg-custom3 py-4 border border-custom1 fs-5"
          type="button"
        >
          +
        </button>
        <button
          class="bg-custom2 col-6 py-4 border border-custom1 fs-5"
          type="button"
        >
          0
        </button>
        <button
          class="bg-custom2 py-4 border border-custom1 fs-5"
          type="button"
        >
          .
        </button>
        <button
          class="bg-custom3 py-4 border border-custom1 fs-5"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing components testing Header.js 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-md navbar-light bg-secondary fixed-top"
      >
        <div
          class="container-fluid"
        >
          <span
            class="navbar-brand text-white"
          >
            Math Magicians
          </span>
          <button
            aria-controls="navbarNav"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-bs-target="#navbarNav"
            data-bs-toggle="collapse"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: white; font-size: 25px;"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </button>
          <div
            class="collapse navbar-collapse"
            id="navbarNav"
          >
            <ul
              class="navbar-nav ms-auto"
            >
              <li
                class="nav-item"
              >
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="v-line d-none d-md-block"
              />
              <li
                class="nav-item"
              >
                <a
                  href="/calculator"
                >
                  Calculator
                </a>
              </li>
              <li
                class="v-line d-none d-md-block"
              />
              <li
                class="nav-item"
              >
                <a
                  href="/quote"
                >
                  Quote
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-md navbar-light bg-secondary fixed-top"
    >
      <div
        class="container-fluid"
      >
        <span
          class="navbar-brand text-white"
        >
          Math Magicians
        </span>
        <button
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNav"
          data-bs-toggle="collapse"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: white; font-size: 25px;"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            class="navbar-nav ms-auto"
          >
            <li
              class="nav-item"
            >
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="v-line d-none d-md-block"
            />
            <li
              class="nav-item"
            >
              <a
                href="/calculator"
              >
                Calculator
              </a>
            </li>
            <li
              class="v-line d-none d-md-block"
            />
            <li
              class="nav-item"
            >
              <a
                href="/quote"
              >
                Quote
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
